packages/assert/src/assert.js

```diff
@@ -95,8 +95,19 @@ function equal(
 ) {
   assert(actual === expected, optDetails);
 }
+
+function assertTypeof(
+  specimen,
+  typename,
+  optDetails = details(['', ` must be ${an(typename)}`], specimen),
```

Comment:
why can't this just be ```optDetails = details`${specimen} must be ${an(typename)}` ```?

Reply:
I'm trying to avoid demanding knowledge of the template API rather than the template usage.

Reply:
Because in the thrown error you wouldn't reveal the contents of `an(typename)` but rather its typeof, which would always be `(a string)` and therefore confusing.

Reply:
That makes me want separate methods then, that could have the string allocated once. Having to concat together a string every time I think I'm just doing a constant compare is the kind of surprise that leads to performance death of a thousand cuts. Or rather, reluctance to use simple type assertions because somehow they are more expensive than less clear ones :)

Reply:
That's a good point I wasn't thinking about. I'll try a different idea tomorrow.

Reply:
@dtribble Take a look at the new `openDetail` declassifier and its uses.